reana_job_controller/job_monitor.py

```diff
@@ -219,6 +221,60 @@ def watch_jobs(self, job_db):
                 time.sleep(120)
 
 
+@singleton
+class JobMonitorSlurmCERN():
+    """Slurm jobs monitor CERN."""
+
+    def __init__(self):
```

Comment:
What do you think about extracting the constructor to a superclass i.e. `JobMonitor` (since the initialisation is the same everywhere) and just passing the `thread_name` as a parameter to it? So we make sure that all monitors are initialised the same way.

Reply:
That would require to add a `NotImplementedError` into `watch_jobs` in the base class, otherwise, the monitor is not going to work (if `watch_jobs` is not implemented).